src/LanguageServer/Impl/Sources/ReferenceSource.cs

```diff
@@ -54,13 +54,19 @@ internal sealed class ReferenceSource {
                 }
 
                 var rootDefinition = definingMember.GetRootDefinition();
-                var name = definingMember.GetName();
-
-                // If it is an implicitly declared variable, such as function or a class
-                // then the location is invalid and the module is null. Use current module.
-                var declaringModule = rootDefinition?.DeclaringModule ?? analysis.Document;
-                if (!string.IsNullOrEmpty(name) && (declaringModule.ModuleType == ModuleType.User || options == ReferenceSearchOptions.All)) {
-                    return await FindAllReferencesAsync(name, declaringModule, rootDefinition, location, definitionSource, cancellationToken);
+                if (rootDefinition != null) {
```

Comment:
This is an overall style nit, but this function looks pretty nested even though it's just a few null checks deep that could probably all return early. `url == null`, `definition == null`, `moduleUri == null`, etc.